modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/SchemaRegistryImpl.java

```diff
@@ -118,7 +118,8 @@ public SchemaDescriptor schema(int ver) {
     /** {@inheritDoc} */
     @Override
     public SchemaDescriptor waitLatestSchema() {
-        int lastVer0 = latestVersionStore.getAsInt();
+        // TODO: remove blocking code https://issues.apache.org/jira/browse/IGNITE-17931
+        int lastVer0 = latestVersionStore.get().join();
```

Comment:
I agree that this is kind of a pointless solution. But I think it is still useful, because blocking code is now in one place, so, when this class is refactored (under a different ticket), we can remove all these blocking calls at once.